Sources/SwiftGraphQLParser/GraphQLParser.swift

```diff
@@ -223,22 +286,22 @@ private extension ArraySlice where Element == Token {
 		}
 		
 		var arguments: [Argument] = []
-		while let argument = try self.readArgument() {
+		while let argument = try self.readArgument(startIndex: startIndex) {
 			arguments.append(argument)
 		}
 		
 		guard arguments.isEmpty == false else {
-			throw InternalParserError(type: .emptyArgumentList, index: start.first!.range.upperBound)
+			throw InternalParserError(type: .emptyArgumentList, errorRange: start.first!.range, contextualStart: startIndex, contextualEnd: self.first?.range.upperBound ?? start.last!.range.upperBound)
```

Comment:
I see this logic for computing the range duplicated a lot here. Maybe refactoring that into a method would make this a bit clearer?

Reply:
Agreed.  I will see if I can get away with a simpler initializer, and if that fails, a function that returns `self.first?.range.upperBound ?? start.last!.range.upperBound`